src/d3a/models/market/two_sided_pay_as_bid.py

```diff
@@ -168,3 +173,58 @@ def accept_bid(self, bid: Bid, energy: float = None,
         if not trade.residual:
             self._notify_listeners(MarketEvent.BID_DELETED, bid=market_bid)
         return trade
+
+    def _perform_pay_as_bid_matching(self):
+        # Pay as bid first
+        # There are 2 simplistic approaches to the problem
+        # 1. Match the cheapest offer with the most expensive bid. This will favor the sellers
+        # 2. Match the cheapest offer with the cheapest bid. This will favor the buyers,
+        #    since the most affordable offers will be allocated for the most aggressive buyers.
+
+        # Sorted bids in descending order
+        sorted_bids = list(reversed(sorted(
```

Comment:
Move these sort operations to a separate method on the base class. That way they can be reused by both classes.